packages/valory/skills/hello_world_abci/rounds.py

```diff
@@ -122,7 +114,7 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         if not self.is_majority_possible(
             self.collection, self.synchronized_data.nb_participants
         ):
-            return self._return_no_majority_event()
+            return self.synchronized_data, Event.NO_MAJORITY
```

Comment:
Same question as above, why is this required?

Reply:
yep, we impose the additional constraint we discussed earlier as well @jmoreira-valory

Reply:
After discussing with @Karrenbelt and reviewing the comment from 1097:

Note that the script will examine the parent classes. In this particular case, it will catch this "NO_MAJORITY" event. So I assume that the refactoring here is **not** to fix a problem in this point of the code. Rather this refactoring is to avoid interpreting that other Rounds return the NO_MAJORITY event (because these rounds will not have that event on their parent class).

This observation applies also to the remaining similar refatorings.